dev -> master 1.6.1

atrigila opened the pull request:
## 1.6.1 - 2025-04-23

### `Added`

### `Changed`

- [#323](https://github.com/nf-core/demultiplex/pull/323) Update `samshee` to 0.2.10.

### `Fixed`

- [#315](https://github.com/nf-core/demultiplex/pull/315) Fix adapter removal tests for various samplesheet formats
- [#316](https://github.com/nf-core/demultiplex/pull/316) Fixed fastp module to properly respect the `trim_fastq` parameter and not run when set to false

atrigila commented:
We should update the snapshots first, in this PR: https://github.com/nf-core/demultiplex/pull/327

maxulysse commented on nextflow.config:
```diff
@@ -220,66 +227,86 @@ env {
+    author          = """Chelsea Sawyer, Edmund Miller, Matthias De Smet""" // The author field is deprecated from Nextflow version 24.10.0, use contributors instead
+    contributors    = [
+        // TODO nf-core: Update the field with the details of the contributors to your pipeline. New with Nextflow version 24.10.0
+        [
+            name: 'Chelsea Sawyer',
+            affiliation: '',
+            email: '',
+            github: '',
+            contribution: [], // List of contribution types ('author', 'maintainer' or 'contributor')
+            orcid: ''
+        ],
+        [
+            name: ' Edmund Miller',
+            affiliation: '',
+            email: '',
+            github: '',
+            contribution: [], // List of contribution types ('author', 'maintainer' or 'contributor')
+            orcid: ''
+        ],
+        [
+            name: ' Matthias De Smet',
+            affiliation: '',
+            email: '',
+            github: '',
+            contribution: [], // List of contribution types ('author', 'maintainer' or 'contributor')
+            orcid: ''
+        ],
```
Can we be more complete on the author/maintainer/contributor list?

atrigila replied:
we have addressed this now!

maxulysse commented on nextflow_schema.json:
```diff
@@ -18,6 +18,7 @@
                 },
                 "skip_tools": {
                     "type": "string",
+                    "pattern": "^(fastp|fastqc|kraken|multiqc|checkqc|falco|md5sum|samshee)(,(fastp|fastqc|kraken|multiqc|checkqc|falco|md5sum|samshee)?,?)*(?<!,)$",
```
There seems to be some duplication in this pattern, I think we can do better.
This should do the trick:
```suggestion
                    "pattern": "^((fastp|fastqc|kraken|multiqc|checkqc|falco|md5sum|samshee)?,?)*(?<!,)$",
```

cf https://github.com/nf-core/sarek/blob/master/nextflow_schema.json#L122

atrigila replied:
Thank you for your suggestion, I added this!

maxulysse reviewed (changes requested):
Minor comments

grst commented on tests/functions/AdapterRemover/test-data/samplesheets/expected_SampleSheet_2.csv:
@edmundmiller should these files be here or in test-datasets?

Personally, I'm not against it, was just wondering if there are any nf-core conventions about having "local" test data.

grst reviewed (approved):
LGTM

atrigila commented:
@maxulysse do you have any idea why the test `latest-stable | docker | 2/6` might be running for 3 hours now?

maxulysse commented:
> @maxulysse do you have any idea why the test `latest-stable | docker | 2/6` might be running for 3 hours now?

Not at all, I'll cancel it and re-trigger

maxulysse commented on .github/workflows/ci.yml:
could we add --verbose on nf-test, so that we have more information about why some tests seem to be stuck

atrigila commented:
@maxulysse  now it works, thank you!

atrigila commented:
We addressed the comments here and the comment regarding the maintainers commented in the previous PR, so if anyone's not opposed, we can go ahead and merge this one to master.
